src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets

```
@@ -133,8 +134,25 @@ Copyright (C) 2012 Xamarin Inc. All rights reserved.
 	</GetReferenceAssemblyPaths>
 </Target>
 
+  <!-- Warn about deprecated configurations.
+       Do it here because we want them to appear on every build,
+       even if we aren't rerunning generator -->
+  <Target Name="_WarnAboutDeprecatedConfigurations">
+    <AndroidWarning Code="XA4229"
+      ResourceName="XA4229"
+      FormatArguments="$(AndroidClassParser)"
+      Condition=" '$(AndroidClassParser)' != 'class-parse' "
+    />
+
+    <AndroidWarning Code="XA4230"
+      ResourceName="XA4230"
+      FormatArguments="$(AndroidCodegenTarget)"
+      Condition=" '$(AndroidCodegenTarget)' != 'XAJavaInterop1' "
+    />
+  </Target>
```

Maybe these should be:
```suggestion
  <Target Name="_WarnAboutDeprecatedConfigurations">
    <AndroidWarning Code="XA4229"
      ResourceName="XA4229"
      FormatArguments="$(AndroidClassParser)"
      Condition=" '$(AndroidClassParser)' == 'jar2xml' "
    />

    <AndroidWarning Code="XA4230"
      ResourceName="XA4230"
      FormatArguments="$(AndroidCodegenTarget)"
      Condition=" '$(AndroidCodegenTarget)' == 'XamarinAndroid' "
    />
  </Target>
```
Just so that if we add a new value for these, we won't accidentally get warnings.

---

I've seen a few cases of people using invalid values like `Xamarin.Android` so this covers them as well. Ideally we eventually just remove the properties completely.  If we later need to bring them back with new values we may end up using different properties.